exec/java-exec/src/main/java/org/apache/drill/exec/ExecConstants.java

```diff
@@ -237,71 +281,81 @@
-      Long.MAX_VALUE, 60 * 1000 * 5);
+  public static final BooleanValidator ENABLE_QUEUE = new BooleanValidator("exec.queue.enable", false);
+  public static final LongValidator LARGE_QUEUE_SIZE = new PositiveLongValidator("exec.queue.large", 1000, 10,
+      "Sets the number of large queries that can run concurrently in the cluster. Range: 0 - 1000.");
+  public static final LongValidator SMALL_QUEUE_SIZE = new PositiveLongValidator("exec.queue.small", 100000, 100,
+      "Sets the number of small queries that can run concurrently in the cluster. Range: 0 - 100000.");
+  public static final LongValidator QUEUE_THRESHOLD_SIZE = new PositiveLongValidator("exec.queue.threshold",
+      Long.MAX_VALUE, 30000000, "Sets the cost threshold, which depends on the complexity of the queries in" +
+      " queue, for determining whether query is large or small. Complex queries have higher thresholds." +
+      " Range: 0 - 9223372036854775807.");
+  public static final LongValidator QUEUE_TIMEOUT = new PositiveLongValidator("exec.queue.timeout_millis",
+      Long.MAX_VALUE, 60 * 1000 * 5, "Indicates how long a query can wait in queue before the query fails." +
+      " Range: 0 - 9223372036854775807.");
 
-  String ENABLE_VERBOSE_ERRORS_KEY = "exec.errors.verbose";
-  OptionValidator ENABLE_VERBOSE_ERRORS = new BooleanValidator(ENABLE_VERBOSE_ERRORS_KEY, false);
+  public static final String ENABLE_VERBOSE_ERRORS_KEY = "exec.errors.verbose";
+  public static final OptionValidator ENABLE_VERBOSE_ERRORS = new BooleanValidator(ENABLE_VERBOSE_ERRORS_KEY, false,
+      "Toggles verbose output of error messages.");
 
-  String ENABLE_NEW_TEXT_READER_KEY = "exec.storage.enable_new_text_reader";
-  OptionValidator ENABLE_NEW_TEXT_READER = new BooleanValidator(ENABLE_NEW_TEXT_READER_KEY, true);
+  public static final String ENABLE_NEW_TEXT_READER_KEY = "exec.storage.enable_new_text_reader";
+  public static final OptionValidator ENABLE_NEW_TEXT_READER = new BooleanValidator(ENABLE_NEW_TEXT_READER_KEY, true);
 
-  String BOOTSTRAP_STORAGE_PLUGINS_FILE = "bootstrap-storage-plugins.json";
-  String MAX_LOADING_CACHE_SIZE_CONFIG = "drill.exec.compile.cache_max_size";
+  public static final String BOOTSTRAP_STORAGE_PLUGINS_FILE = "bootstrap-storage-plugins.json";
+  public static final String MAX_LOADING_CACHE_SIZE_CONFIG = "drill.exec.compile.cache_max_size";
 
-  String DRILL_SYS_FILE_SUFFIX = ".sys.drill";
+  public static final String DRILL_SYS_FILE_SUFFIX = ".sys.drill";
 
-  String ENABLE_WINDOW_FUNCTIONS = "window.enable";
-  OptionValidator ENABLE_WINDOW_FUNCTIONS_VALIDATOR = new BooleanValidator(ENABLE_WINDOW_FUNCTIONS, true);
+  public static final String ENABLE_WINDOW_FUNCTIONS = "window.enable";
+  public static final OptionValidator ENABLE_WINDOW_FUNCTIONS_VALIDATOR = new BooleanValidator(ENABLE_WINDOW_FUNCTIONS,
+      true, "Enable or disable window functions in Drill 1.1 and later.");
```

Comment:
No one who uses Drill 1.0 will see this message, so leave out "in Drill 1.1 and later"